src/core/flow.js

```diff
@@ -498,6 +505,44 @@ export function Flow({ moves, phases, endIf, turn, events, plugins }) {
     return state;
   }
 
+  function UpdateStage(state, { arg, playerID }) {
+    if (!arg) return state;
+    if (typeof arg === 'string') {
+      arg = { stage: arg };
+    }
+
+    let { ctx } = state;
+    let {
+      activePlayers,
+      _activePlayersMoveLimit,
+      _activePlayersNumMoves,
+    } = ctx;
+
+    if (arg.stage) {
+      if (activePlayers === null) {
+        activePlayers = {};
+      }
+      activePlayers[playerID] = arg.stage;
+      _activePlayersNumMoves[playerID] = 0;
```

Comment:
Do we need to set this if a `moveLimit` is not defined?

Reply:
When implementing `moveLimit`, I decided to always increment `_activePlayersNumMoves` in `ProcessMove` even when `moveLimit` is not enabled, which means it would need resetting for new stages.

https://github.com/nicolodavis/boardgame.io/blob/297eb87433ce165d5ff231ef7485a05e1192d797/src/core/flow.js#L695

It isn’t used by the framework if `moveLimit` is not defined, but I thought it might be helpful for people who want to keep track of moves without using `moveLimit`.

Reply:
OK.